pkg/oc/admin/policy/modify_roles.go

```diff
@@ -470,22 +509,54 @@ func (o *RoleModificationOptions) AddRole() error {
 	roleBinding.RoleRef.Namespace = o.RoleNamespace
 	roleBinding.RoleRef.Name = o.RoleName
 
+	// warn if binding to non-existent role
+	if o.RoleAccessor != nil && o.PrintErrf != nil {
+		if _, err := o.RoleAccessor.GetRole(roleBinding.RoleRef.Name); err != nil {
+			if kapierrors.IsNotFound(err) {
+				o.PrintErrf("Warning: role '%s' not found\n", roleBinding.RoleRef.Name)
+			} else {
+				o.PrintErrf("Warning: role '%s' may not exist; unknown error: %v\n", roleBinding.RoleRef.Name, err)
+			}
+		}
+	}
 	newSubjects := authorizationapi.BuildSubjects(o.Users, o.Groups)
 	newSubjects = append(newSubjects, o.Subjects...)
-
 subjectCheck:
 	for _, newSubject := range newSubjects {
+		if o.PrintErrf != nil {
+			var err error
+			switch newSubject.Kind {
+			case authorizationapi.ServiceAccountKind:
+				if o.ServiceAccountClient != nil {
+					_, err = o.ServiceAccountClient.ServiceAccounts(newSubject.Namespace).Get(newSubject.Name, metav1.GetOptions{})
+				}
+			case authorizationapi.UserKind, authorizationapi.SystemUserKind:
+				if o.UserClient != nil {
+					_, err = o.UserClient.User().Users().Get(newSubject.Name, metav1.GetOptions{})
```

Comment:
this would make almost all users of this command get a warning on every invocation. the vast majority of users do not have read access to cluster-scoped User objects.

Reply:
these lookups also would not interact well with external authentication configurations (e.g. bearer token webhook configs) where an external system is the source of user info

Reply:
Yeah, the unauthorized error case is problematic.
For non-admin users, getting a warning every time is unnecessary noise and provides no additional benefit, as per your comment.
But not printing the warning would be an inconsistent behavior for admins switching between contexts. They might incorrectly assume the user (or group/sa/role) exists when they don't get a warning, not knowing the difference in behavior between levels of access.

As for the external auth configurations, I think the warning still provides value. It shows that at least the user has not yet logged in (which may or may not be expected).